concrete-core-fixture/src/generation/synthesizing/glwe_ciphertext.rs

```diff
@@ -63,4 +101,223 @@ mod backend_core {
+                    unsafe {
+                        Vec::from_raw_parts(slice.as_ptr() as *mut u32, slice.len(), slice.len())
+                    },
+                    polynomial_size,
+                )
+                .unwrap();
+            ProtoBinaryGlweCiphertext32(ciphertext)
+        }
+
+        fn destroy_glwe_ciphertext_view(
+            &mut self,
+            entity: GlweCiphertextView32,
+            drop_underlying_memory: bool,
+        ) {
+            // We don't know if this the last view referring to the slice it contains, so let the
+            // programmer tell us if that's the case.
+            if drop_underlying_memory {
+                // If requested re-construct the vector so that it frees memory when it's dropped
+                let slice = self
+                    .core_engine
+                    .consume_retrieve_glwe_ciphertext(entity)
+                    .unwrap();
+                unsafe {
+                    Vec::from_raw_parts(slice.as_ptr() as *mut u32, slice.len(), slice.len())
+                };
+            } else {
+                self.core_engine.destroy(entity).unwrap();
+            }
+        }
+    }
+
+    impl<'a> SynthesizesGlweCiphertextView<Precision64, GlweCiphertextView64<'a>> for Maker {
+        fn synthesize_glwe_ciphertext_view(
+            &mut self,
+            prototype: &Self::GlweCiphertextProto,
+        ) -> GlweCiphertextView64<'a> {
+            let ciphertext = prototype.0.to_owned();
+            let polynomial_size = ciphertext.polynomial_size();
+            let container = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(ciphertext)
+                .unwrap();
+            self.core_engine
+                .create_glwe_ciphertext(container.leak() as &[u64], polynomial_size)
+                .unwrap()
+        }
+
+        fn unsynthesize_glwe_ciphertext_view(
+            &mut self,
+            entity: GlweCiphertextView64,
+        ) -> Self::GlweCiphertextProto {
+            let polynomial_size = entity.polynomial_size();
+            let slice = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(entity)
+                .unwrap();
+            let ciphertext = self
+                .core_engine
+                .create_glwe_ciphertext(
+                    unsafe {
+                        Vec::from_raw_parts(slice.as_ptr() as *mut u64, slice.len(), slice.len())
+                    },
+                    polynomial_size,
+                )
+                .unwrap();
+            ProtoBinaryGlweCiphertext64(ciphertext)
+        }
+
+        fn destroy_glwe_ciphertext_view(
+            &mut self,
+            entity: GlweCiphertextView64,
+            drop_underlying_memory: bool,
+        ) {
+            // We don't know if this the last view referring to the slice it contains, so let the
+            // programmer tell us if that's the case.
```

Comment:
I don't understand this choice sorry :slightly_frowning_face: Why can't we simply call the destroy? This part is still very confusing for me sorry! :sweat_smile:

Reply:
See above

Reply:
this won't be a problem with the reworked design thanks to Alex, resolving